import/src/test/java/com/google/cloud/healthcare/imaging/dicomadapter/backupuploader/GcpBackupUploaderTest.java

```diff
@@ -0,0 +1,266 @@
+        exceptionRule.expect(GcpBackupUploader.GcpUriParseException.class);
+        exceptionRule.expectMessage("Invalid upload path");
+
+        new GcpBackupUploader(UPLOAD_PATH_SPACE_PROGECT_NAME);
+    }
+
+    @Test
+    public void parseUri_Failed_OnInvalidBucketName() throws GcpBackupUploader.GcpUriParseException {
+        exceptionRule.expect(GcpBackupUploader.GcpUriParseException.class);
+        exceptionRule.expectMessage("Invalid upload path");
+
+        new GcpBackupUploader(UPLOAD_PATH_EMPTY_BUCKET_NAME);
+    }
+
+    @Test
+    public void parseUri_Failed_OnSpaceBucketName() throws GcpBackupUploader.GcpUriParseException {
+        exceptionRule.expect(GcpBackupUploader.GcpUriParseException.class);
+        exceptionRule.expectMessage("Invalid upload path");
+
+        new GcpBackupUploader(UPLOAD_PATH_SPACE_BUCKET_NAME);
+    }
+
+    @Test
+    public void parseUri_Failed_OnInvalidUploadObject() throws GcpBackupUploader.GcpUriParseException {
+        exceptionRule.expect(GcpBackupUploader.GcpUriParseException.class);
+        exceptionRule.expectMessage("Invalid upload path");
+
+        new GcpBackupUploader(UPLOAD_PATH_EMPTY_UPLOAD_OBJECT);
+    }
+
+    @Test
+    public void parseUri_Failed_OnSpaceUploadObject() throws GcpBackupUploader.GcpUriParseException {
+        exceptionRule.expect(GcpBackupUploader.GcpUriParseException.class);
+        exceptionRule.expectMessage("Invalid upload path");
+
+        new GcpBackupUploader(UPLOAD_PATH_SPACE_UPLOAD_OBJECT);
+    }
+
+    @Test
+    public void getCredential() throws IOException {
+        Credentials creds = gcpBackupUploader.getCredential(ENV_CREDS);
+        System.out.println(creds.getAuthenticationType());
+
+        assertThat(creds).isInstanceOf(Credentials.class);
+        assertThat(creds).isNotNull();
+        assertThat(creds.getAuthenticationType()).isEqualTo(AUTH_TYPE);
+    }
+
+    @Test
+    public void getCredential_Failed_OnEmptyEnv() throws IOException {
+        exceptionRule.expect(NullPointerException.class);
+        gcpBackupUploader.getCredential("");
+    }
+
+    @Test
+    public void getCredential_Failed_OnNotExistsEnv() throws IOException {
+        exceptionRule.expect(NullPointerException.class);
+        gcpBackupUploader.getCredential(NOT_EXISTS_ENV);
+    }
+
+    @Test
+    public void getCredential_Failed_OnInvalidEnv() throws IOException {
+        exceptionRule.expect(NullPointerException.class);
+        gcpBackupUploader.getCredential(INVALID_ENV);
+    }
+
+    @Test
+    public void getUploadFilePath() {
+        assertThat(gcpBackupUploader.getUploadFilePath()).isNotNull();
+        assertThat(gcpBackupUploader.getUploadFilePath()).isNotEmpty();
+        assertThat(gcpBackupUploader.getUploadFilePath()).isEqualTo(UPLOAD_PATH);
+    }
+
+    @Test
+    public void doWriteBackup() throws IBackupUploader.BackupException {
+        gcpBackupUploader.doWriteBackup(BYTES_SEQ, UNIQ_NAME);
```

Comment:
Unit test can`t do real calls to GCP. GCP have util classes and practices for isolated testing of features. This class may help: https://googleapis.dev/java/google-cloud-clients/0.90.0-alpha/com/google/cloud/storage/contrib/nio/testing/LocalStorageHelper.html